### ISSUE TITLE
Slap over SSH on new Ubuntu server gives error: "Cannot find module"

### ISSUE THREAD

**Issue by kenmorechalfant** [slap-editor/slap#293]
I have just made a new VPS on DigitalOcean, Ubuntu 14.04 LEMP stack.
Installed Node v5.7.0.

Ran
`sudo npm install -g slap@latest`

And when I try to edit a file I get the error: `Cannot find module '../build/Release/pathwatcher.node'`

My Slap version info is:
`slap@0.1.60, node@5.7.0, v8@4.6.85.31, http_parser@2.6.2, uv@1.8.0, zlib@1.2.8, ares@1.10.1-DEV, icu@56.1, modules@47, openssl@1.0.2f`

The full error I get is:
```
module.js:341
    throw err;
    ^

Error: Cannot find module '../build/Release/pathwatcher.node'
    at Function.Module._resolveFilename (module.js:339:15)
    at Function.Module._load (module.js:290:25)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (/usr/lib/node_modules/slap/node_modules/pathwatcher/lib/main.js:6:13)
    at Object.<anonymous> (/usr/lib/node_modules/slap/node_modules/pathwatcher/lib/main.js:213:4)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (/usr/lib/node_modules/slap/node_modules/text-buffer/lib/text-buffer.js:11:10)
    at Object.<anonymous> (/usr/lib/node_modules/slap/node_modules/text-buffer/lib/text-buffer.js:1406:4)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (/usr/lib/node_modules/slap/node_modules/editor-widget/lib/Editor.js:8:18)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (/usr/lib/node_modules/slap/node_modules/editor-widget/index.js:1:80)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
    at require (internal/module.js:16:19)
    at Object.<anonymous> (/usr/lib/node_modules/slap/lib/ui/Slap.js:12:14)
    at Module._compile (module.js:413:34)
    at Object.Module._extensions..js (module.js:422:10)
    at Module.load (module.js:357:32)
    at Function.Module._load (module.js:314:12)
    at Module.require (module.js:367:17)
```

**Comment by kenmorechalfant** [comment]
OK so here's how I solved my problem.
First I `npm uninstall`ed slap (not sure if necessary).

Then I changed to root user (also not sure if necessary), and tried to install again, still got errors but this time I noticed it said something about not having 'make' installed. I installed make and ran it again. It got further but I noticed another error that "command: g++" not found. So I googled that and found a forum post saying to try:

`apt-get install build-essential g++`

And that did the trick.

Went over to my other server and had a different problem; NPM would just quit half way and say `killed`. Googled that, realized I was running out of memory -> created a swapfile and voila. It works there too.
